### DISCUSSION TITLE
What is the difference between Eclipse Theia and Coder Code Server?

### DISCUSSION THREAD

**Discussion by viveksh1** [eclipse-theia/theia#9431] [Q&A]
What is the difference between Eclipse Theia and Coder Code Server?

**Comment by vince-fugnitto** [comment]
@VladPutinEu thanks for the discussion, do you have any specific questions, or just want want an overall differences?

**Eclipse Theia** is a framework to build IDE-like applications for both the `browser` and `desktop`. The framework permits application developers to create their custom vendor neutral applications, which suit their needs and use-cases, can be customized, and modular. 
The framework makes use of vscode components when necessary (such as the monaco-editor) and supports the vscode extension API. The framework also supports the language server protocol (LSP), debug adapter protocol (DAP), trace server protocol (TSP) as well. 

From what I understand `coder` is an application in itself (not a framework to build products) of the open-source components of vscode, patched for the browser. Due to `theia` being a framework, application developers can extend, or modify the default behavior of their application, something that is not permissible with an actual application like `coder` as you will likely need to fork, maintain and uplift your application if you want to customize its internals which becomes increasingly difficult over time.

You may also find additional information here:
- https://github.com/eclipse-theia/theia/issues/5347#issuecomment-498385901
- https://eclipsesource.com/blogs/2019/12/06/the-eclipse-theia-ide-vs-vs-code/
